msal/authority.py

```diff
@@ -58,11 +58,11 @@ def __init__(self, authority_url, validate_authority=True,
         _, _, self.tenant = canonicalize(self.token_endpoint)  # Usually a GUID
         self.is_adfs = self.tenant.lower() == 'adfs'
 
-    def user_realm_discovery(self, username):
+    def user_realm_discovery(self, username, headers):
```

Comment:
If we ended up really need to add telemetry for this User Realm Discovery behaviour (see the comment above), then let's put the headers as an optional parameter with default value as `None`. Because this header thing is NOT part of this method's core business.